### ISSUE TITLE
ProGuard configuration parser error : proguard.txt line 12:21 no viable alternative at input ','

### ISSUE THREAD

**Issue by wezley98** [urbanairship/android-library#93]
### What Urban Airship dependencies are you using?
  implementation "com.urbanairship.android:urbanairship-sdk:9.5.0 

### What are the versions of any relevant development tools you are using?
Android Studio 3.1.4

# Report

### What unexpected behavior are you seeing?
Can't build with proguard turned off :
 
useProguard false
minifyEnabled true

* What went wrong:
Execution failed for task ':app:transformClassesWithAndroidGradleClassShrinkerForUkDebugLive'.
> ProGuard configuration parser error: /Users/xxxxxx/.gradle/caches/transforms-1/files-1.1/urbanairship-core-9.5.0.aar/fab21318f63828b4d76d217480eec362/proguard.txt line 12:21 no viable alternative at input ','

### What is the expected behavior?
Builds run with the following settings:

useProguard false
minifyEnabled true

**Comment by marc-scig** [comment]
Hi @wezley98, 

It looks as though the shrinker feature might be incompatible with the "includedescriptorclasses" annotation we recently added to a subset of our proguard rules. We're looking into potential solutions, but in the meanwhile, what happens if you try to override those rules with something like this:

```
## Autopilot
-keep public class com.urbanairship.Autopilot
-keep public class * extends com.urbanairship.Autopilot

## Push Providers
-keep public class * extends com.urbanairship.push.PushProvider
-keepclassmembernames class * extends com.urbanairship.push.PushProvider {
  public <methods>;
  public <fields>;
}
```

Here's the rules the SDK is currently using, for comparison: https://github.com/urbanairship/android-library/blob/master/urbanairship-core/proguard-rules.pro

**Comment by wezley98** [comment]
@marc-scig 
Thanks that appears to be working for now.

**Comment by wezley98** [comment]
@marc-scig Sorry my earlier comment was wrong, overwriting the proguard rules doesn't fix the issue. Waiting on a new release I guess?

**Comment by marc-scig** [comment]
Hi @wezley98,

Thanks for the feedback. I've been looking into this and I was able to reproduce your build error by using these settings, and that particular error went away after removing "includedescriptorclasses" from the proguard rules, as expected. However, this introduced additional problems for our Firebase dependencies at runtime, resulting in a NoClassDefFoundError at start. I was hoping to put out a fix for this ASAP but we need to do a deeper investigation to figure out how to properly support this feature or whether that's even possible at this time.

For what it's worth, Android labels this feature as "experimental" in their documentation, and our support for such things is tentative depending on the level of stability we can guarantee. In the meanwhile, if it's possible I'd recommend turning off shrinking for debug while we figure out how to do this the right way.

**Comment by rlepinski** [comment]
Fixed in 9.5.2. Thanks for the report.
